clap_complete/src/dynamic/shells/mod.rs

```diff
@@ -27,6 +107,70 @@ pub enum CompleteCommand {
 }
 
 /// Generally used via [`CompleteCommand`]
+///
+/// This is the arguments for the `complete` subcommand.
+///
+/// **NOTE**: We should replace the `completion.bash` to filename we wanted, also `completion.fish`, `completion.zsh`, `completion.ps1`, `completion.elv`.
+///
+/// To generate shell completion scripts, we can use the `complete` subcommand as follows:
+///
+/// - Bash
+/// 	- `your_program complete --shell bash --register /path/to/completion.bash`
```

Comment:
We should not be encouraging people to write out the completions like this as they can go stale on upgrade.

Reply:
After getting the completion script from the `complete` subcommand and outputting it to stdout, which way do you prefer: 
- putting the whole script directly into the shell configuration file.
- saving the script in a separate file and then loading it from the shell configuration file? 

Personally, I prefer the second method. Just a heads-up, we won't be putting the script in a system directory to avoid issues when upgrading.

Reply:
For bash, you could do
```bash
source <(your_program complete --shell bash --register -)
```
No intermediate files needed.